src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -0,0 +1,1024 @@
+
+/**
+ * Converts between bytes and chars and hex-encoded strings which may include additional
+ * formatting markup such as prefixes, suffixes, and delimiters.
+ * <p>
+ * There are two factories of {@code HexFormat} with preset parameters {@link #of()} and
+ * {@link #ofDelimiter(String) of(delimiter)}. For other parameter combinations
+ * the {@code withXXX} methods return copies of {@code HexFormat} modified
+ * {@link #withPrefix(String)}, {@link #withSuffix(String)}, {@link #withDelimiter(String)}
+ * or choice of {@link #withUpperCase()} or {@link #withLowerCase()} parameters using
+ * a fluent builder style.
+ * <p>
+ * For primitive to hexadecimal string conversions the {@code toHexDigits}
+ * methods include {@link #toHexDigits(byte)}, {@link #toHexDigits(int)}, and
+ * {@link #toHexDigits(long)}, etc.
+ * For conversions producing uppercase hexadecimal strings use {@link #withUpperCase()}.
+ *
+ * <p>
+ * For hexadecimal string to primitive conversions the {@code fromHexDigits}
+ * methods include {@link #fromHexDigits(CharSequence) fromHexDigits(string)},
+ * {@link #fromHexDigitsToLong(CharSequence) fromHexDigitsToLong(string)}, and
+ * {@link #fromHexDigit(int) fromHexDigit(int)} converts a single character or codepoint.
+ *
+ * <p>
+ * For byte array to formatted hexadecimal string conversions
+ * the {@code formatHex} methods include {@link #formatHex(byte[]) formatHex(byte[])}
+ * and {@link #formatHex(Appendable, byte[]) formatHex(Appendable, byte[])}.
+ * The formatted output can be appended to {@link StringBuilder}, {@link System#out},
+ * {@link java.io.Writer}, and {@link java.io.PrintStream}, all of which are {@link Appendable}.
+ * Each byte value is formatted as the prefix, two hexadecimal characters from the
+ * uppercase or lowercase digits, and the suffix.
+ * A delimiter appears after each formatted value, except the last.
+ * For conversions producing uppercase hexadecimal strings use {@link #withUpperCase()}.
+ *
+ * <p>
+ * For formatted hexadecimal string to byte array conversions the
+ * {@code parseHex} methods include {@link #parseHex(CharSequence) parseHex(string)} and
+ * {@link #parseHex(char[], int, int) parseHex(char[], offset, length)}.
+ * Each byte value is parsed as the prefix, two hexadecimal characters from the
+ * uppercase or lowercase digits, and the suffix.
+ * The delimiter is required after each formatted value, except the last.
+ *
+ * @apiNote
+ * For example, an individual byte is converted to a string of hexadecimal digits using
+ * {@link HexFormat#toHexDigits(int) toHexDigits(int)} and converted from a string to a
+ * primitive value using {@link HexFormat#fromHexDigits(CharSequence) fromHexDigits(string)}.
+ * <pre>{@code
+ *     HexFormat hex = HexFormat.of();
+ *     byte b = 127;
+ *     String byteStr = hex.toHexDigits(b);
+ *
+ *     int byteVal = hex.fromHexDigits(byteStr);
+ *     assert(byteStr.equals("7f"));
+ *     assert(b == byteVal);
+ *
```

Comment:
I understand why this is the case, but the example is highlighting that a roundtrip from byte is not possible :-(

Reply:
In the string form, the value is unsigned and not known to be a byte.
The example should reinforce that point by being explicit about casting the value back to a byte.
```
            byte byteVal = (byte)hex.fromHexDigits(byteStr);
            assert(byteStr.equals("7f"));
            assert(b == byteVal);```